releng/buildpacks/java/README.md

```diff
@@ -1,73 +1,77 @@
+    --docker-username=<your-username> \
+    --docker-password=<your-password> \
+    --docker-server=https://index.docker.io/v1/ \
+    --namespace default
+```
+
+
+1. Create Service Account
+```
+kubectl apply -f service-account.yaml
 ```
 
-Build neo-builder (include `unzip` used in the neo-java-buildpack)
-```bash
-cd neo-builder
-pack builder create neo-builder:bionic --config ./builder.toml
+1. Build `Kneo XSK Stack`:
+
 ```
+cd stack/
 
+docker build . -t dirigiblelabs/neo-stack-base --target base
+docker push dirigiblelabs/neo-stack-base
 
-Build neo app with `neo-java-buildpack`
-```bash
-cd cki
-pack build neo-test --builder neo-builder:bionic --buildpack ./neo-java-buildpack --path <path-to-war-directory> --network host
+docker build . -t dirigiblelabs/neo-stack-run --target run
+docker push dirigiblelabs/neo-stack-run
+
+docker build . -t dirigiblelabs/neo-stack-build --target build
+docker push dirigiblelabs/neo-stack-build
 ```
 
-## Neo persistence application example
-
-Prerequisites:
-- Hana instance
-
-Go to https://tools.hana.ondemand.com/ and download the `neo-java-web-sdk-3.127.13.3.zip` and unzip it.
-
-```bash
-# Navigate to sample app
-cd neo-java-web-sdk-3.127.13.3/samples/persistence-with-jdbc
-# Create DB resource. In NEO this is provided by the runtime.
-cat > WebContent/META-INF/context.xml <<EOF
-<Context>
-  <Resource name="jdbc/DefaultDB" auth="Container" type="javax.sql.DataSource"
-               maxTotal="100" maxIdle="30" maxWaitMillis="10000"
-               username="<username>" password="<password>" driverClassName="com.sap.db.jdbc.Driver"
-               url="jdbc:sap://<server>:<port>/<database>"/>
-</Context>
-EOF
-# build war
-mvn clean install
-# build app with buildpacks
-pack build neo-test --builder neo-builder:bionic --buildpack ~/workspace/neo-buildpack/cki/neo-java-buildpack --path target/ --network host
-# run app 
-docker run -p 8080:8080 neo-test
+1. Build `Kneo Buildpack`:
+
 ```
-Visit http://localhost:8080/persistence-with-jdbc/ to see if everything works fine.
+cd buildpack/
 
-## Notes
-To improve dev efficiency, you can run:
-```bash
-# download neo-runtime to cwd
-wget http://nexus.wdf.sap.corp:8081/nexus/content/groups/build.milestones/com/sap/cloud/runtime/kotyo/com.sap.cloud.runtime.kotyo/2.174.1/com.sap.cloud.runtime.kotyo-2.174.1.zip
+pack buildpack package dirigiblelabs/neo-buildpack --config ./package.toml
+docker push dirigiblelabs/neo-buildpack
+```
+
+1. Create `ClusterStore`, `ClusterStack` and `Builder`:
+
+```
+cd kpack/
+
+kubectl apply -f kpack.yaml
+```
+
+1. Create Image:
+
+```yaml
+apiVersion: kpack.io/v1alpha1
+kind: Image
+metadata:
+    name: tutorial-image
+    namespace: default
+spec:
+    tag: dirigiblelabs/neo-test
+    serviceAccount: tutorial-service-account
+    builder:
+        name: neo-builder
+        kind: Builder
+    source:
+        blob:
+            url: https://github.com/SAP/xsk/raw/main/modules/basis/basis-datasource/samples/test_db.war
```

Comment:
Please move the sample `.war` file into the `samples` folder.

Reply:
I can move it but package `basis-datasource/samples` depends on it and we will have to change the context of its Dockerfile. Are you ok with this ?